tools/notarize

```diff
@@ -43,22 +43,26 @@ rcodesign sign \
   "${BINARY}"
 echo "Signed ${BINARY}!"
 
+# Currently, NOTARIZE is always set to 0. In effect, this means that the
+# binary is signed, but not notarized or stapled. This is because notarizing
+# requires a DMG which is difficult to properly create on Linux. Since all of 
+# the official installation processes are through brew or wget, this will not
+# be a problem.
+#
+# Where this does become a problem is if a user wants to install from a browser.
+# In this situation, unless there will need to be a notarization process otherwise 
+# MacOS will mark our binary as unsafe and the user will need to go through hoops
+# to install it.
 if [[ "${NOTARIZE}" == "1" ]]; then
-  which mkfs.hfsplus || sudo apt-get install hfsprogs -y
+  which zip || sudo apt-get install zip -y
```

Comment:
The comment above indicates that it is not possible to notarize the `zip` file, but then the changes indicate that it may be possible?

Would it make sense to leave the `DMG` stuff in place? It seems more likely that there would be a way to notarize the `DMG` before Apple accepts notarization of a `zip`.

Reply:
My comment is misleading in this sense. [You can notarize a `zip` file](https://developer.apple.com/documentation/security/notarizing_macos_software_before_distribution/customizing_the_notarization_workflow#:~:text=While%20you%20can%20notarize%20a%20ZIP%20archive%2C%20you%20can%E2%80%99t%20staple%20to%20it%20directly.), I'll make that comment clearer.

That being said, the old DMG logic wasn't working due to us not "properly" creating a DMG on Linux. So if we ever came back to use that code the DMG creation would not work. Thoughts?